src/transformStateWithClones.js

```diff
@@ -65,7 +65,35 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, transforms) {
-  // write code here
+  let initialState = { ...state };
+  const changesList = [];
+
+  if (transforms.length === 0) {
+    return changesList;
+  }
+
+  for (const transform of transforms) {
+    switch (transform.operation) {
+      case 'clear':
+        initialState = {};
+        changesList.push({ ...initialState });
+        break;
+
+      case 'addProperties':
+        Object.assign(initialState, transform.properties);
+        changesList.push({ ...initialState });
+        break;
+
+      case 'removeProperties':
+        for (const key of transform.properties) {
+          delete initialState[key];
+        }
+        changesList.push({ ...initialState });
+        break;
+    }
+  }
```

Comment:
you can push changed state at the end of each loop

```suggestion
  for (const transform of transforms) {
    switch (transform.operation) {
      case 'clear':
        initialState = {};
        break;

      case 'addProperties':
        Object.assign(initialState, transform.properties);
        break;
        
      case 'removeProperties':
        for (const key of transform.properties) {
          delete initialState[key];
        }
        break;
    }
    
     changesList.push({ ...initialState });
  }
```